### PR TITLE
PLAT-419: Add the deploy preview reusable workflow

### PR DESCRIPTION
**Pull request by bjarneo**

### CODE REVIEW

File: .github/workflows/deploy-preview.yaml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+      # ===  Get the latest git short sha
+      - name: Get the git short id
+        id: short_id
+        run: |
+          sha=$(cut -c 1-7 <<< $GITHUB_SHA)
+          echo "::set-output name=sha::$sha"
+
+      # === github.head_ref has to be used to get the branch name while in PR state
+      - name: Branch name to URL friendly string
```
Comment (bjarneo) [inline .github/workflows/deploy-preview.yaml:71]: Create an URL friendly action / or find one

File: .github/workflows/deploy-preview.yaml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        uses: docker/build-push-action@v2
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: ${{ inputs.app_name }}
+        with:
+          push: true
+          cache-from: type=registry,ref=${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.BRANCH_NAME }}
+          cache-to: type=inline
+          build-args: ${{ inputs.docker_build_args }}
+          tags: |
```
Comment (bjarneo) [inline .github/workflows/deploy-preview.yaml:97]: Add tag with latest, and update the cache-from with that postfix

File: .github/workflows/deploy-preview.yaml [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+          version: "latest"
+        id: kubectl
+
+      # === This will run a kubectl request to 
+      # 1. fetch the index of the preview host to check the existence
+      # 2. if the host do not exist, patch the ingress with the new host
+      - name: Add the service to the preview ingress
+        shell: bash
+        run: |
+          EXIST=$(${{ steps.kubectl.outputs.kubectl-path }} get ingress preview-ingress -n preview -o json  | jq '.spec.rules | map(.host == "${{ env.BRANCH_NAME }}.preview.dignio.dev") | index(true)')
```
Comment (bjarneo) [inline .github/workflows/deploy-preview.yaml:130]: use the env var for the namespace

Comment (bjarneo) [reply .github/workflows/deploy-preview.yaml:130]: replace short arguments with long ones (i.e. -n -> --namespace)

File: .github/workflows/deploy-preview.yaml [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+          # If the preview ingress does not have the host name, add it.
+          if [ "$EXIST" == "null" ] 
+          then 
+            echo "Creating a preview URL."
+
+            ${{ steps.kubectl.outputs.kubectl-path }} patch ingress preview-ingress -n preview --type='json' -p='[ { "op":"add","path":"/spec/rules/-", "value": { "host": "${{ env.BRANCH_NAME }}.preview.dignio.dev", "http": { "paths": [ { "backend": { "service": { "name": "${{ inputs.app_name }}-${{ env.BRANCH_NAME }}-preview", "port": { "number": ${{ inputs.port }} } } }, "path": "${{ inputs.path }}", "pathType": "Prefix" } ] } } } ]'
```
Comment (bjarneo) [inline .github/workflows/deploy-preview.yaml:137]: sue env var + replace short arguments

**Review by lemonsaurus — APPROVED** [review]
🐦 Did you know, the word parakeet means "long tail". 🐦

Also this PR looks good.

**Pull request merged**